x-pack/apm-server/sampling/eventstorage/storage_manager.go

```diff
@@ -341,30 +397,64 @@ func (sm *StorageManager) WriteSubscriberPosition(data []byte) error {
 // NewUnlimitedReadWriter returns a read writer with no storage limit.
 // For testing only.
```

Comment:
nit: Not a fan of these testing-only methods, can we move at least this one to the test file?

Reply:
The test file is in another package though, moving this means making the struct field public which is even worse imo. The one with least impact to prod code is to make the test file use the same package, which is probably the best option

Reply:
Hmm, isn't `NewReadWriter` a public method? I was thinking of moving the `newUnlimitedReadWriter` to the test package.

Reply:
oh, sorry i thought you were talking about `WithGetDiskUsage`.

Yes, UnlimitedReadWriter can be moved, but quite a few test files use it. I can sort it out.